lib/cylc/network/server.py

```diff
@@ -84,12 +85,13 @@ def __init__(self, encode_method, decode_method, secret_method):
         self.secret = secret_method
 
     def start(self, min_port, max_port):
-        """Start the server running
+        """Start the server running.
 
-        Args:
-            ports (iterable): Generator of ports (int) to choose from.
-                The lowest available port will be chosen.
+        Will use a port range provided to select random ports.
 
+        Args:
+            min_port (int): minimum socket port number
+            max_port (int): maximum socket port number
         """
         # create socket
         self.context = zmq.Context()
```

Comment:
I think the context should probably be set in the `__init__` method. *(Looking at this code again the separation of `init` and `start` is purely for separation of concerns and not really functional, calling `start` twice would be nasty.)*

Reply:
:+1: good spot! Thanks!